project/Build.scala

```diff
@@ -889,11 +891,28 @@ object Build {
           baseDirectory.value.getParentFile.getParentFile / "shared/src/test/scala-ide-stubs"
       } else {
         sourceGenerators in Test += Def.task {
+          val s = streams.value
+          val log = s.log
+          val lm = dependencyResolution.value
+          val binVer = scalaBinaryVersion.value
+
+          val retrieveDir = s.cacheDirectory / "previous-stdlibs"
+
+          val previousStdLibs = previousVersions.map { version =>
+            val jars = lm.retrieve("org.scala-js" % s"scalajs-library_$binVer" % version intransitive(),
+                scalaModuleInfo = None, retrieveDir, log)
+              .fold(w => throw w.resolveException, _.distinct)
+            assert(jars.size == 1, jars.toString())
+            version -> jars.head
+          }.toMap
+
```

Comment:
This is cargo-culted form `linkerInterfaceJS`. Do you see any de-duplication opportunities?

Reply:
The only thing I see would be to define a helper
```scala
def retrieveSingleArtifact(moduleID: ModuleID, lm: DependencyResolution,
    retrieveDir: File, log: ManagedLogger): File = {
  val jars = lm.retrieve(moduleID.intransitive(), scalaModuleInfo = None, retrieveDir, log)
    .fold(w => throw w.resolveException, _.distinct)
  assert(jars.size == 1, jars.toString())
  jars.head
}
```
and call that from here and `linkerInterfaceJS`. Not sure whether it's worth it.